route-rs-runtime/src/link/sync_link.rs

```diff
@@ -1,22 +1,73 @@
 use crate::element::Element;
-use crate::link::PacketStream;
+use crate::link::{ElementLinkBuilder, Link, LinkBuilder, PacketStream};
 use futures::{Async, Poll, Stream};
 
-pub struct SyncLink<E: Element> {
-    input_stream: PacketStream<E::Input>,
+#[derive(Default)]
+pub struct SyncLinkBuilder<E: Element> {
+    in_streams: Option<Vec<PacketStream<E::Input>>>,
+    element: Option<E>,
+}
+
+impl<E: Element> SyncLinkBuilder<E> {
+    pub fn new() -> Self {
+        SyncLinkBuilder {
+            in_streams: None,
+            element: None,
+        }
+    }
+}
+
+/// Although `Link` allows an arbitrary number of ingressors and egressors, `SyncLink`
+/// may only have one ingress and egress stream since it lacks some kind of queue
+/// storage. In the future we might decide to restrict the interface for this link
+/// for clearer intent.
+impl<E: Element + Send + 'static> LinkBuilder<E::Input, E::Output> for SyncLinkBuilder<E> {
+    fn ingressors(self, in_streams: Vec<PacketStream<E::Input>>) -> Self {
```

Comment:
Do we want to enforce that each builder method should be called at most once? For example, we could assert here that `self.in_streams` must be `None`, otherwise panic with a helpful message.

Reply:
I think the question is: Does running the function twice cause issues other than the obvious  'The values change' answer.  If yes, then we should prevent double running.

Reply:
I mean, no it shouldn't, but we should be able to quickly detect nonsensical builders like:

```
SyncLinkBuilder
  .ingressors(<some vec>)
  .ingressors(<wait... another vec?>)
  .element(e)
  .build_link();
```

I guess running the function twice causes the issue of "being confusing to the reader". Functionally the last builder value is used.

Reply:
Hmm, if you want to make a double call panic, I am fine with that.